src/main/java/org/example/StudentsGroup.java

```diff
@@ -0,0 +1,59 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class StudentsGroup {
+    private Student groupLeader;
+    private List<Student> studentsList;
+    private List<String> tasksList;
+    private Map<Student, List<Integer>> completedTasks;
+
+    public StudentsGroup(Student groupLeader) {
+        this.groupLeader = groupLeader;
+        this.studentsList = new ArrayList<>();
+        this.tasksList = new ArrayList<>();
+        this.completedTasks = new HashMap<>();
+    }
+
+    public void changeGroupLeader(Student newLeader) {
+        this.groupLeader = newLeader;
```

Comment:
Also on changing group leader check that it's already in students list or no, if no then add